homeassistant/components/motioneye/switch.py

```diff
@@ -67,38 +84,37 @@ class MotionEyeSwitch(MotionEyeEntity, SwitchEntity):
 
     def __init__(
         self,
+        entity_description: EntityDescription,
         config_entry_id: str,
         camera: dict[str, Any],
-        switch_key: str,
-        switch_key_friendly_name: str,
         client: MotionEyeClient,
         coordinator: DataUpdateCoordinator,
         options: MappingProxyType[str, str],
-        enabled_by_default: bool,
     ) -> None:
         """Initialize the switch."""
-        self._switch_key = switch_key
-        self._switch_key_friendly_name = switch_key_friendly_name
+        self._entity_description = entity_description
         super().__init__(
             config_entry_id,
-            f"{TYPE_MOTIONEYE_SWITCH_BASE}_{switch_key}",
+            f"{TYPE_MOTIONEYE_SWITCH_BASE}_{self._entity_description.key}",
```

Comment:
This is a local variable

```suggestion
            f"{TYPE_MOTIONEYE_SWITCH_BASE}_{entity_description.key}",
```